### PR TITLE
Drop support for sw < 6.5, drop support for php8.1, add support for s…

### PR DESCRIPTION
**Pull request by jkniest**
## Info

We are currently preparing a larger release (v3) with a few breaking changes. Meaning we will merge multiple stuff with breaking changes into main until we tag a new release (and write upgrade guides, etc.)

### CODE REVIEW

File: src/Command/LoadFixtureGroupCommand.php [hunk 1]
```diff
@@ -5,30 +5,25 @@
...
+    public function __construct(
+        private readonly FixtureLoader $loader
+    ) {
+        parent::__construct();
     }
 
     protected function configure(): void
     {
-        $this
-            ->setHelp('This command allows you to run a group of fixtures')
```
Comment (CreativLul) [inline src/Command/LoadFixtureGroupCommand.php:30]: Is it intentional that the help has been removed here?

File: src/Command/LoadFixturesCommand.php [hunk 2]
```diff
@@ -33,6 +30,6 @@ protected function execute(InputInterface $input, OutputInterface $output): int
 
         $io->success('Done!');
 
-        return 0;
+        return Command::SUCCESS;
```
Comment (CreativLul) [inline src/Command/LoadFixturesCommand.php:33]: What happens if there are issues during the execution of the command. Is it correct to always return 'success'?

File: src/Command/LoadSingleFixtureCommand.php [hunk 3]
```diff
@@ -5,30 +5,26 @@
...
+    public function __construct(
+        private readonly FixtureLoader $loader
+    ) {
+        parent::__construct();
     }
 
     protected function configure(): void
     {
-        $this
-            ->setHelp('This command allows you to run only one specific fixture')
```
Comment (CreativLul) [inline src/Command/LoadSingleFixtureCommand.php:30]: Is it intentional that the help has been removed here?

File: src/Utils/CategoryUtils.php [hunk 4]
```diff
@@ -1,5 +1,7 @@
 <?php
 
+/** @noinspection ALL */
```
Comment (CreativLul) [inline src/Utils/CategoryUtils.php:3]: Should this stay in here?

File: src/Utils/CategoryUtils.php [hunk 5]
```diff
@@ -10,13 +12,11 @@
...
-class CategoryUtils
+readonly class CategoryUtils
 {
-    private EntityRepository $categoryRepository;
-
-    public function __construct(EntityRepository $categoryRepository)
-    {
-        $this->categoryRepository = $categoryRepository;
+    public function __construct(
+        private EntityRepository $categoryRepository
```
Comment (rvalley98) [inline src/Utils/CategoryUtils.php:18]: readonly

File: src/Utils/CmsUtils.php [hunk 6]
```diff
@@ -11,13 +11,11 @@
...
-class CmsUtils
+readonly class CmsUtils
 {
-    private EntityRepository $cmsPageRepository;
-
-    public function __construct(EntityRepository $cmsPageRepository)
-    {
-        $this->cmsPageRepository = $cmsPageRepository;
+    public function __construct(
+        private EntityRepository $cmsPageRepository
```
Comment (rvalley98) [inline src/Utils/CmsUtils.php:17]: readonly

File: src/Utils/CustomerUtils.php [hunk 7]
```diff
@@ -10,13 +10,11 @@
...
-class CustomerUtils
+readonly class CustomerUtils
 {
-    private EntityRepository $salutationRepository;
-
-    public function __construct(EntityRepository $salutationRepository)
-    {
-        $this->salutationRepository = $salutationRepository;
+    public function __construct(
+        private EntityRepository $salutationRepository
```
Comment (rvalley98) [inline src/Utils/CustomerUtils.php:16]: readonly

File: src/Utils/MediaUtils.php [hunk 8]
```diff
@@ -12,38 +12,14 @@
...
-    {
-        $this->mediaRepository       = $mediaRepository;
-        $this->mediaFolderRepository = $mediaFolderRepository;
-        $this->fileSaver             = $fileSaver;
-        $this->fileFetcher           = $fileFetcher;
+    public function __construct(
+        private EntityRepository $mediaRepository,
+        private EntityRepository $mediaFolderRepository,
+        private FileSaver $fileSaver,
+        private FileFetcher $fileFetcher
```
Comment (rvalley98) [inline src/Utils/MediaUtils.php:21]: private readonly 
```suggestion
        private readonly EntityRepository $mediaRepository,
        private readonly EntityRepository $mediaFolderRepository,
        private readonly FileSaver $fileSaver,
        private readonly FileFetcher $fileFetcher
```

File: src/Utils/PaymentMethodUtils.php [hunk 9]
```diff
@@ -11,25 +11,11 @@
...
-     *
-     * @param EntityRepository $paymentMethodRepository
-     *
-     * @noinspection PhpMissingParamTypeInspection We can not use typehints until PaymentMethodRepositoryDecorator is removed (Shopware 6.5).
-     */
-    public function __construct($paymentMethodRepository)
-    {
-        $this->paymentMethodRepository = $paymentMethodRepository;
+    public function __construct(
+        private EntityRepository $paymentMethodRepository
```
Comment (rvalley98) [inline src/Utils/PaymentMethodUtils.php:17]:
```suggestion
        private readonly EntityRepository $paymentMethodRepository
```

File: src/Utils/SalesChannelUtils.php [hunk 10]
```diff
@@ -17,25 +17,17 @@
...
-        $this->currencyRepository     = $currencyRepository;
-        $this->localeRepository       = $localeRepository;
+    public function __construct(
+        private EntityRepository $salesChannelRepository,
+        private EntityRepository $snippetSetRepository,
+        private EntityRepository $taxRepository,
+        private EntityRepository $countryRepository,
+        private EntityRepository $languageRepository,
+        private EntityRepository $currencyRepository,
+        private EntityRepository $localeRepository
```
Comment (rvalley98) [inline src/Utils/SalesChannelUtils.php:29]:
```suggestion
        private EntityRepository $salesChannelRepository,
        private readonly EntityRepository $snippetSetRepository,
        private readonly EntityRepository $taxRepository,
        private readonly EntityRepository $countryRepository,
        private readonly EntityRepository $languageRepository,
        private readonly EntityRepository $currencyRepository,
        private EntityRepository $localeRepository
```

File: src/Utils/ShippingMethodUtils.php [hunk 11]
```diff
@@ -10,13 +10,11 @@
...
-class ShippingMethodUtils
+readonly class ShippingMethodUtils
 {
-    private EntityRepository $shippingMethodRepository;
-
-    public function __construct(EntityRepository $shippingMethodRepository)
-    {
-        $this->shippingMethodRepository = $shippingMethodRepository;
+    public function __construct(
+        private EntityRepository $shippingMethodRepository
```
Comment (rvalley98) [inline src/Utils/ShippingMethodUtils.php:16]:
```suggestion
        private readonly EntityRepository $shippingMethodRepository
```

File: src/Utils/CustomerUtils.php (hunk 7 above)
Comment (rvalley98) [inline src/Utils/CustomerUtils.php:16]:
```suggestion
        private readonly EntityRepository $salutationRepository
```

File: src/Utils/CmsUtils.php (hunk 6 above)
Comment (rvalley98) [inline src/Utils/CmsUtils.php:17]:
```suggestion
        private readonly EntityRepository $cmsPageRepository
```

File: src/Utils/CategoryUtils.php (hunk 5 above)
Comment (rvalley98) [inline src/Utils/CategoryUtils.php:18]:
```suggestion
        private readonly EntityRepository $categoryRepository
```

File: src/FixtureHelper.php [hunk 12]
```diff
@@ -12,25 +12,17 @@
...
-        $this->shippingMethodUtils = $shippingMethodUtils;
-        $this->customerUtils       = $customerUtils;
+    public function __construct(
+        private MediaUtils $mediaUtils,
+        private CategoryUtils $categoryUtils,
+        private SalesChannelUtils $salesChannelUtils,
+        private CmsUtils $cmsUtils,
+        private PaymentMethodUtils $paymentMethodUtils,
+        private ShippingMethodUtils $shippingMethodUtils,
+        private CustomerUtils $customerUtils
```
Comment (rvalley98) [inline src/FixtureHelper.php:24]:
```suggestion
        private readonly MediaUtils $mediaUtils,
        private readonly CategoryUtils $categoryUtils,
        private readonly SalesChannelUtils $salesChannelUtils,
        private readonly CmsUtils $cmsUtils,
        private readonly PaymentMethodUtils $paymentMethodUtils,
        private readonly ShippingMethodUtils $shippingMethodUtils,
        private readonly CustomerUtils $customerUtils
```

**Review by CreativLul — APPROVED** [review]
Looking good

File: src/Command/LoadFixtureGroupCommand.php (hunk 1 above)
Comment (jkniest) [reply src/Command/LoadFixtureGroupCommand.php:30]: Yes, I've added the help text to the description in the attribute on top of the class:

```
#[AsCommand('fixture:load:group', 'Run a specific fixture group')]
```

Also I've shortened the text

File: src/Command/LoadSingleFixtureCommand.php (hunk 3 above)
Comment (jkniest) [reply src/Command/LoadSingleFixtureCommand.php:30]: Yes, I've added the help text to the description in the attribute on top of the class

File: src/Utils/MediaUtils.php (hunk 8 above)
Comment (jkniest) [reply src/Utils/MediaUtils.php:21]: I've made the whole class readonly, which all attributes automatically inherit:

CleanShot 2024-04-05 at 15 34 00@2x

File: src/Utils/PaymentMethodUtils.php (hunk 9 above)
Comment (jkniest) [reply src/Utils/PaymentMethodUtils.php:17]: See other comment regarding readonly classes :)

File: src/Utils/SalesChannelUtils.php (hunk 10 above)
Comment (jkniest) [reply src/Utils/SalesChannelUtils.php:29]: See other comment regarding readonly classes :)

File: src/Utils/ShippingMethodUtils.php (hunk 11 above)
Comment (jkniest) [reply src/Utils/ShippingMethodUtils.php:16]: See other comment regarding readonly classes :)

File: src/Utils/CustomerUtils.php (hunk 7 above)
Comment (jkniest) [reply src/Utils/CustomerUtils.php:16]: See other comment regarding readonly classes :)

File: src/Utils/CmsUtils.php (hunk 6 above)
Comment (jkniest) [reply src/Utils/CmsUtils.php:17]: See other comment regarding readonly classes :)

File: src/Utils/CategoryUtils.php (hunk 5 above)
Comment (jkniest) [reply src/Utils/CategoryUtils.php:18]: See other comment regarding readonly classes :)

File: src/FixtureHelper.php (hunk 12 above)
Comment (jkniest) [reply src/FixtureHelper.php:24]: See other comment regarding readonly classes :)

File: src/Command/LoadFixturesCommand.php (hunk 2 above)
Comment (jkniest) [reply src/Command/LoadFixturesCommand.php:33]: Nope :-D I've refatored it in this PR https://github.com/basecom/FixturesPlugin/pull/54

Maybe it is okay, to keep it there for now, sicne both MRs will be released together in a new version

File: src/Utils/CategoryUtils.php (hunk 4 above)
Comment (jkniest) [reply src/Utils/CategoryUtils.php:3]: Removed it, thanks!

**Review by rvalley98 — APPROVED** [review]
Nice work

**Pull request merged**